scipy/optimize/tests/test_bracket.py

```diff
@@ -179,7 +179,7 @@ def f(*args, **kwargs):
             ref_attr = [xp.asarray(getattr(ref, attr)) for ref in refs]
             res_attr = getattr(res, attr)
             xp_assert_close(xp_ravel(res_attr, xp=xp), xp.stack(ref_attr))
-            xp_assert_equal(res_attr.shape, shape)
+            assert res_attr.shape == shape
```

Comment:
It was passing because the scipy variant of `array_namespace()` applied to a tuple returns numpy:
https://github.com/scipy/scipy/blob/e1a841b8beecfb14daf250490302354aef776746/scipy/_lib/_array_api.py#L87-L89

So `_assert_matching_namespace` was observing that numpy both for the actual and the desired array.